mycroft/client/speech/hotword_factory.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2017 Mycroft AI, Inc.
+#
+# This file is part of Mycroft Core.
+#
+# Mycroft Core is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Mycroft Core is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Mycroft Core.  If not, see <http://www.gnu.org/licenses/>.
+
+from mycroft.configuration import ConfigurationManager
+from mycroft.util.log import getLogger
+from os.path import dirname, exists, join, abspath
+import os
+import time
+import tempfile
+
+__author__ = 'seanfitz, jdorleans, jarbas'
+
+LOG = getLogger("HotwordFactory")
+
+BASEDIR = dirname(abspath(__file__)).join("recognizer")
```

Comment:
If you are going to append "recognizer" here, call it something like `RECOGNIZER_DIR` for clarity.